src/adagrams.js

```diff
@@ -1,15 +1,100 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
-};
+  let alphabetPool = 'aaaaaaaaaabbccddddeeeeeeeeeeeeffggghhiiiiiiiiijkllllmmnnnnnnooooooooppqrrrrrrssssttttttuuuuvvwwxyyz'
+  alphabetPool = alphabetPool.toUpperCase()
+  let i = 0;
+  let playerHand = []
+  
+  do {
+    i++;
+    let singleLetter = alphabetPool.charAt(Math.random() * alphabetPool.length);
+    playerHand.push(singleLetter);
+    alphabetPool = alphabetPool.replace(singleLetter, '');
+  } while (i < 10);
+
+  return playerHand;
+}
 
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let playerHand = {}
+  for (let i of lettersInHand) {
+    if (i in playerHand) {
+      playerHand[i] += 1
+    } else {
+      playerHand[i] = 1
+    }
+  } 
+
+  for (let j of input) {
+    if (!(j in playerHand)) {
+      return false;
+    } else {
+      playerHand[j] -= 1
+    }
+  }
+
+  return Object.values(playerHand).every(
+    value => value >= 0
+  )
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+  const scoreChart = {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z']
+  };
+  
+  let wordScore = 0;
+  for (let i = 0; i < word.length; i++) {
+    for (const [key, value] of Object.entries(scoreChart)) {
+      for (const v of value) {
+        if (word[i].toUpperCase() === v) {
+          wordScore += Math.floor(key)
+        }
+      }
+    }
+  };
```

Comment:
Instead of using three `for loops` here you could simplify this function by looping over the word & checking if the letter is included in the `alphabetPool` like so:
```
  for (let i = 0; i < word.length; i++) {
    let letter = word[i];
    // Add the score for the letter to the total score
    if (alphabetPool.includes(letter)) {
      score += letterValues[letter];
    }
  }
```